compiler/rustc_metadata/src/creader.rs

```diff
@@ -558,10 +560,11 @@ impl<'a, 'tcx> CrateLoader<'a, 'tcx> {
                     dep_kind = CrateDepKind::MacrosOnly;
                 }
                 data.update_dep_kind(|data_dep_kind| cmp::max(data_dep_kind, dep_kind));
+                data.update_private_dep(|private_dep| private_dep && private_dep);
```

Comment:
This line does nothing. Did you mean this?

```suggestion
                data.update_private_dep(|p_d| p_d && private_dep);
```